src/main/java/core/basesyntax/ColorSupplier.java

```diff
@@ -1,7 +1,16 @@
 package core.basesyntax;
 
-public class ColorSupplier {
+import java.util.Random;
+
+public class ColorSupplier extends Ball {
```

Comment:
The ColorSupplier class should not extend the Ball class. It doesn't represent a 'is-a' relationship. A ColorSupplier is not a type of Ball, it's a utility class that provides a color for a Ball.